internal/controllers/machine/machine_controller.go

```diff
@@ -525,7 +525,8 @@ func (r *Reconciler) drainNode(ctx context.Context, cluster *clusterv1.Cluster,
 			log.Error(err, "Could not find node from noderef, it may have already been deleted")
 			return ctrl.Result{}, nil
 		}
-		return ctrl.Result{}, errors.Errorf("unable to get node %q: %v", nodeName, err)
```

Comment:
I think we have to make sure that those error messages don't change on every reconcile. Otherwise we get a lot of updates on the Machine CR

Reply:
Is this a change in this PR? We were already returning the error string, just not correctly wrapped.

Reply:
Oh my bad, missed it

Reply:
sorry to waste your time ,I think I'd better close it and firstly fix my develop env

Reply:
It's a good point though and it can be easily missed, and a pretty major bug if it gets through. We should probably have some sort of sanitizer for errors which end up in our conditions right? I wonder is it possible to do that in a generic way? The main thing is to avoid adding object names where we know they're variable - or at least that's the issue I recall having before.

Reply:
Hard to say if there's a good way. But the nodeName is definitely not variable here :)

Reply:
> Hard to say if there's a good way. But the nodeName is definitely not variable here :)
> 
> I was just wondering about the error, but as you said if there's an issue it wouldn't be a new one

I don't quite understand what you mean

Reply:
I am saying that when the reconciler reconciles a single machine the nodeName will be the same every time.
So that won't lead to different errors / different conditions on the Machine / infinite reconciles.

I know know how the `err` looks like and if it's the same every time. But we included the error message already before this PR so this PR won't introduce a bug. If there is a bug it was already there before

Reply:
Yes, errs may be the same until the error is fixed.Other errs in reconciles can be the same, right?